### PR TITLE
[TACHYON-1566] Refine LoginUser clean or re-login methods in security tests

### PR DESCRIPTION
**Pull request by dongche**
JIRA: https://tachyon.atlassian.net/browse/TACHYON-1566

In current test package, some tests about security feature need to clean or re-login the LoginUser. We could provide util methods for these test
* resetLoginUser() to clean the singleton
* updateLoginUser() to clean and re-login a new user atomically. This could avoid flaky test results.

@apc999 , @sundapeng

### CODE REVIEW

File: tests/src/test/java/tachyon/security/LoginUserTestUtils.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  }
+
+  /**
+   * Resets the {@link LoginUser} and re-login with new user
+   *
+   * @param conf the instance of {@link TachyonConf}
+   * @param user the new user
+   * @throws IOException if login fails
+   */
+  public static void updateLoginUser(TachyonConf conf, String user) throws IOException {
```
Comment (yupeng9) [inline tests/src/test/java/tachyon/security/LoginUserTestUtils.java]: What do you think of merging these two methods into one? When user is null, it behaves as reset.

Comment (yupeng9) [inline tests/src/test/java/tachyon/security/LoginUserTestUtils.java]: Or have the same method name but different number of parameters.

File: tests/src/test/java/tachyon/shell/AbstractTfsShellTest.java [hunk 2]
```diff
@@ -80,7 +79,7 @@ public final void after() throws Exception {
 
   @Before
   public final void before() throws Exception {
-    Whitebox.setInternalState(LoginUser.class, "sLoginUser", (String) null);
+    cleanLoginUser();
```
Comment (yupeng9) [inline tests/src/test/java/tachyon/shell/AbstractTfsShellTest.java]: clean -> clear?

**Comment by yupeng9** [comment]
Thanks for the improvement. I left some comments.

**Comment by dongche** [comment]
@yupeng9 Thanks for your review. The methods are renamed.

**Comment by yupeng9** [comment]
Please resolve the conflicts.

**Comment by dongche** [comment]
Conflicts resolved.

**Comment by apc999** [comment]
+1 LGTM

**Comment by yupeng9** [comment]
LGTM

**Pull request merged**